### PR TITLE
[TEVA-2852] vacancy expired more than year publisher dashboard message

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-2582

Screenshot 2021-09-07 at 10 23 55

### CODE REVIEW

File: app/views/publishers/vacancies/job_applications/_banner.html.slim [hunk 1]
```diff
@@ -26,7 +26,7 @@
                 = t(".deadline.before")
             span.govuk-body = format_time_to_datetime_at(vacancy.expires_at)
 
-      - unless vacancy.draft?
+      - unless vacancy.draft? || vacancy.expires_at < 1.year.ago
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/_banner.html.slim:29]: Please either use the scope `created_within_data_access_period` or the constant `DATA_ACCESS_PERIOD_FOR_PUBLISHERS`

File: app/views/publishers/vacancies/job_applications/_banner.html.slim [hunk 2]
```diff
@@ -47,5 +47,8 @@
     = render(DashboardComponent.new) do |dashboard|
       - dashboard.navigation_item item: (govuk_link_to t("buttons.manage_listing"), organisation_job_path(vacancy.id), class: "dashboard-component-navigation__link", "aria-current": ("page" if controller_name == "vacancies"))
       - if vacancy.can_receive_job_applications? && !current_organisation.local_authority?
-        - dashboard.navigation_item item: (govuk_link_to t(".applications", count: vacancy.job_applications.where(status: %w[submitted reviewed shortlisted unsuccessful]).count), organisation_job_job_applications_path(vacancy.id), class: "dashboard-component-navigation__link", "aria-current": ("page" if controller_name == "job_applications"))
+        - if vacancy.expires_at < 1.year.ago
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/_banner.html.slim:50]: See above

File: app/views/publishers/vacancies/job_applications/index.html.slim [hunk 3]
```diff
@@ -2,19 +2,20 @@
...
-.govuk-grid-row class="govuk-!-margin-top-7"
-  .govuk-grid-column-full
-    = form_for sort_form, as: "", url: organisation_job_job_applications_path(vacancy.id), method: :get, data: { "auto-submit": true, "hide-submit": true } do |f|
-      = f.govuk_collection_select :sort_column,
-        sort,
-        :column,
-        :display_name,
-        label: { text: t("jobs.sort_by.label"), size: "s", class: "govuk-label inline govuk-!-margin-right-2" },
-        selected: :last
+- if vacancy.expires_at > 1.year.ago && job_applications.any?
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/index.html.slim:5]: Same

File: app/views/publishers/vacancies/job_applications/index.html.slim [hunk 4]
```diff
@@ -2,19 +2,20 @@
...
+          sort,
+          :column,
+          :display_name,
+          label: { text: t("jobs.sort_by.label"), size: "s", class: "govuk-label inline govuk-!-margin-right-2" },
+          selected: :last
 
 .govuk-grid-row
   .govuk-grid-column-full
-    - if job_applications.any?
+    - if job_applications.any? && vacancy.expires_at > 1.year.ago
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/index.html.slim:18]: Same!

File: app/views/publishers/vacancies/job_applications/index.html.slim [hunk 5]
```diff
@@ -37,5 +38,9 @@
                   = t("buttons.reject")
                   span.govuk-visually-hidden = " #{application.name}"
 
+    - elsif vacancy.expires_at < 1.year.ago
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/index.html.slim:41]: S'me

File: app/views/publishers/vacancies/job_applications/_banner.html.slim (hunk 1 above)
Comment (david-mears-dfe) [reply app/views/publishers/vacancies/job_applications/_banner.html.slim:29]: Or write a predicate method on vacancy

Comment (alexbowen) [reply app/views/publishers/vacancies/job_applications/_banner.html.slim]: have used the constant

File: app/views/publishers/vacancies/job_applications/index.html.slim [hunk 6]
```diff
@@ -37,5 +38,9 @@
                   = t("buttons.reject")
                   span.govuk-visually-hidden = " #{application.name}"
 
+    - elsif vacancy.expires_at < DATA_ACCESS_PERIOD_FOR_PUBLISHERS
```
Comment (david-mears-dfe) [inline app/views/publishers/vacancies/job_applications/index.html.slim:41]: Can't this just be `else`?

**Comment by david-mears-dfe** [comment]
Ticket: https://dfedigital.atlassian.net/browse/TEVA-2852

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**